### PR TITLE
AArch64: Implement iabs/labsEvaluator()

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements iabs/labsEvaluator() for aarch64.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
These evaluators are used in `comptest`.

File: compiler/aarch64/codegen/UnaryEvaluator.cpp [hunk 1]
```diff
@@ -81,17 +81,36 @@ TR::Register *OMR::ARM64::TreeEvaluator::lnegEvaluator(TR::Node *node, TR::CodeG
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+static TR::Register *commonAbsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/UnaryEvaluator.cpp]: Can you call this `commonIntegerAbsEvaluator` to make it clear this is for integer operations?

File: compiler/aarch64/codegen/UnaryEvaluator.cpp [hunk 2]
```diff
@@ -81,17 +81,36 @@ TR::Register *OMR::ARM64::TreeEvaluator::lnegEvaluator(TR::Node *node, TR::CodeG
...
+   {
+   TR::Register *trgReg = cg->allocateRegister();
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *srcReg = cg->evaluate(firstChild);
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/UnaryEvaluator.cpp]: Could this be a clobberEvaluate to save a register?  I think you still need a temp register, but the target and source could potentially be shared.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 3]
```diff
@@ -343,10 +343,11 @@ OMR::ARM64::TreeEvaluator::checkcastAndNULLCHKEvaluator(TR::Node *node, TR::Code
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::treetopEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::treetopEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: can you split the treetopEvaluator into a separate commit?  It's not related to abs.

Comment (knn-k) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: I separated treetopEvaluator() as #3151.

**Comment by knn-k** [comment]
Updated files reflecting the review comments.

File: compiler/aarch64/codegen/UnaryEvaluator.cpp [hunk 4]
```diff
@@ -81,17 +81,35 @@ TR::Register *OMR::ARM64::TreeEvaluator::lnegEvaluator(TR::Node *node, TR::CodeG
...
 	}
 
+static TR::Register *commonIntegerAbsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *reg = cg->gprClobberEvaluate(firstChild);
+   TR::Register *tempReg = cg->allocateRegister();
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic eorOp = is64bit ? TR::InstOpCode::eorx : TR::InstOpCode::eorw;
```
Comment (andrewcraik) [inline compiler/aarch64/codegen/UnaryEvaluator.cpp]: While this doesn't need to be fixed here, it might be worth considering some helpers somewhere in the opcode implementation which will return the right opcode for an operation based on a width. We have some of those in the x86 where you can ask for a load or other operator for a given width which centralizes the opcode selection and reduces the ternary operators in the codegen code. It also provides a central place to update opcode choices later if needed. Again, more of a suggestion as the aarch64 implementation moves forward rather than saying this code itself should not be accepted.

Comment (0xdaryl) [reply compiler/aarch64/codegen/UnaryEvaluator.cpp]: I agree with this suggestion.  It's been handy on x86 for sure.  @knn-k , please open an issue for this so it can get on the backlog.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

File: compiler/aarch64/codegen/UnaryEvaluator.cpp (hunk 4 above)
Comment (knn-k) [reply compiler/aarch64/codegen/UnaryEvaluator.cpp]: Opened Issue #3269.
